templates/oauth2/oauth2-login.html.twig

```diff
@@ -0,0 +1,144 @@
+                </div>
+                {% endif %}
+
+                {% if enforce_signin_email %}
+                <div class="form-group">
+                    <input class="form-control" type="hidden" value="{{ user.email|attr }}">
+                </div>
+                {% endif %}
+                <div class="form-group">
+                    <input class="form-control" type="hidden" name="username" value="{{ user.username|attr }}">
+                </div>
+                <div class="form-group">
+                    <input class="form-control" type="hidden" name="password" value="{{ user.password|attr }}">
+                </div>
+                <input class="form-control" type="hidden" name="mfa_type" value="TOTP">
+            {% endif %}
+            <hr />
+            <div class="row">
+                <div class="col-md-12">
+                    <div class="btn-group">
+                        {% if mfaRequired is empty %}
+                            <button type="submit" name="user_role" class="btn btn-outline-primary" value="api">{{"OpenEMR Login"|xlt }} <i class="fa fa-sign-in-alt"></i></button>
+                            {% if patientRoleSupport is not empty %}
+                                <button type="submit" name="user_role" class="btn btn-outline-info" value="portal-api">{{ "Patient Login"|xlt  }} <i class="fa fa-sign-in-alt"></i></button>
+                            {% endif %}
+                        {% endif %}
+                    </div>
+                    <div class="form-check-inline float-right">
+                        <input class="form-check-input" type="checkbox" name="persist_login" id="persist_login" value="1">
+                        <label for="persist_login" class="form-check-label">{{ "Remember Me"|xlt  }}</label>
+                    </div>
+                </div>
+            </div>
+        </form>
+    </div>
+</div>
+<script>
+    function doAuth() {
+        let xl = window.top.xl || function(id) { return id; };
+        var f = document.getElementById("userLogin");
+        var requests = JSON.parse(f.form_requests.value);
+        // The server's getAuthenticateData() repeats the same challenge in all requests.
+        var challenge = requests[0].challenge;
+        var registeredKeys = new Array();
+        for (var i = 0; i < requests.length; ++i) {
+            registeredKeys[i] = {"version": requests[i].version, "keyHandle": requests[i].keyHandle};
+        }
+        u2f.sign({{ appId|json_encode }}, challenge, registeredKeys,
+            function (data) {
+                if (data.errorCode && data.errorCode != 0) {
+                    alert(xl("Key access failed with error") +' ' + data.errorCode);
+                    return;
+                }
+                //hide totp input if both on used
+                if (document.getElementById('totp_token')) {
+                    document.getElementById('totp_token').style.display = 'none';
+                }
```

Comment:
```suggestion
                const totpTokenElement = document.getElementById('totp_token');
                if (totpTokenElement) {
                    totpTokenElement.style.display = 'none';
                }
```
To avoid querying the document twice, you can save the result in a variable

Reply:
Fixed.